dso-l1/src/main/java/com/tc/object/InFlightMonitor.java

```diff
@@ -39,6 +39,14 @@ public InFlightMonitor(MessageCodec<?, R> codec, InvokeMonitor<R> monitor, Execu
     this.executor = executor;
   }
 
+  public void exception(EntityException ee) {
+    if (executor != null) {
+      executor.execute(()->monitor.exception(ee));
+    } else {
+      monitor.exception(ee);
+    }
+  }
+
```

Comment:
Probably need to add something for results as well.

Reply:
Same as above: `InFlightMonitor` now discriminates between sync and async.